src/daft-core/src/array/ops/cast.rs

```diff
@@ -48,109 +51,6 @@ use {
-        let source_arrow_array = match source_dtype {
-            // Wrapped primitives
-            DataType::Decimal128(..)
-            | DataType::Date
-            | DataType::Timestamp(..)
-            | DataType::Duration(..)
-            | DataType::Time(..) => {
-                with_match_daft_logical_primitive_types!(source_dtype, |$T| {
-                    use arrow2::array::Array;
-                    to_cast
-                        .physical
-                        .data()
-                        .as_any()
-                        .downcast_ref::<arrow2::array::PrimitiveArray<$T>>()
-                        .unwrap()
-                        .clone()
-                        .to(source_arrow_type)
-                        .to_boxed()
-                })
-            }
-            _ => cast(
-                to_cast.physical.data(),
-                &source_arrow_type,
-                CastOptions {
-                    wrapped: true,
-                    partial: false,
-                },
-            )?,
-        };
-
-        // Then, cast source Arrow LogicalArray to target Arrow LogicalArray.
-
-        cast(
-            source_arrow_array.as_ref(),
-            &target_arrow_type,
-            CastOptions {
-                wrapped: true,
-                partial: false,
-            },
-        )?
-    };
-
-    // If the target type is also Logical, get the Arrow Physical.
-    let result_arrow_physical_array = {
-        if dtype.is_logical() {
-            let target_physical_type = dtype.to_physical().to_arrow()?;
-            match dtype {
-                // Primitive wrapper types: change the arrow2 array's type field to primitive
-                DataType::Decimal128(..)
-                | DataType::Date
-                | DataType::Timestamp(..)
-                | DataType::Duration(..)
-                | DataType::Time(..) => {
-                    with_match_daft_logical_primitive_types!(dtype, |$P| {
-                        use arrow2::array::Array;
-                        result_arrow_array
-                            .as_any()
-                            .downcast_ref::<arrow2::array::PrimitiveArray<$P>>()
-                            .unwrap()
-                            .clone()
-                            .to(target_physical_type)
-                            .to_boxed()
-                    })
-                }
-                _ => cast(
-                    result_arrow_array.as_ref(),
-                    &target_physical_type,
-                    CastOptions {
-                        wrapped: true,
-                        partial: false,
-                    },
-                )?,
-            }
-        } else {
-            result_arrow_array
-        }
-    };
-
-    let new_field = Arc::new(Field::new(to_cast.name(), dtype.clone()));
-    Series::from_arrow(new_field, result_arrow_physical_array)
-}
-
 fn arrow_cast<T>(to_cast: &DataArray<T>, dtype: &DataType) -> DaftResult<Series>
```

Comment:
I see that `arrow_cast` is only used now in `impl DataArray<T> {fn cast(...) {...}}`

Shall we inline it into the implementation of `DataArray` then? Hopefully this would discourage future abuse of this functionality.